alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java

```diff
@@ -149,7 +159,21 @@ public void startup(CoreContainer coreContainer)
      */
     void setupNewDefaultCores(String names)
     {
+        setupNewDefaultCores(names, 1, 1, 1, 1, null);
+    }
 
+    /**
+     * Creates new default cores based on the "createDefaultCores" String passed in.
+     *
+     * @param names comma delimited list of core names that will be created.
+     * @param numShards The total number of shards.
+     * @param replicationFactor - Not sure why the core needs to know this.
+     * @param nodeInstance - Not sure why the core needs to know this.
+     * @param numNodes - Not sure why the core needs to know this.
```

Comment:
We could add this however i'm not clear on the benefits of having n numbers of shard in the same node.

Reply:
@tpage-alfresco, I have a couple of questions please.
- Can these param names be the same as the ones that are set in solrcore.properties? e.g. shard.instance, shard.count, to avoid confusion?
- Also, how would replicationFactor work?
- This would work (for now) with the default sharding method = DBID and could be extended later with additional properties required for the other sharding methods (as part of a separate jira in search backlog) Is that right?

Reply:
I've removed the three parameters I couldn't see the point of. Meenal and I talked about the parameter naming and it turns out that we already have inconsistency. I've left the names as they are, but I don't have a strong preference about this.  The two naming conventions are documented in these two docs pages:

`numShards` and `shardIds`: https://docs.alfresco.com/6.0/concepts/dynamic-sharding.html

`shard.count` and `shard.instance`: https://docs.alfresco.com/5.2/concepts/solr6-shard-approaches.html

As a side note, I'm not sure if `shard.instance` allows a comma separated list of ids, but `shardIds` does.

Reply:
To expand on this last point, we can achieve the configuration from [this diagram](https://docs.alfresco.com/sites/docs.alfresco.com/files/public/images/docs/default6_0/shardtable.png) with the following options:

```
-Dcreate.alfresco.defaults=alfresco,archive -Dnum.shards=12 -Dshard.ids=0,1,6,7
-Dcreate.alfresco.defaults=alfresco,archive -Dnum.shards=12 -Dshard.ids=1,2,7,8
-Dcreate.alfresco.defaults=alfresco,archive -Dnum.shards=12 -Dshard.ids=2,3,8,9
-Dcreate.alfresco.defaults=alfresco,archive -Dnum.shards=12 -Dshard.ids=3,4,9,10
-Dcreate.alfresco.defaults=alfresco,archive -Dnum.shards=12 -Dshard.ids=4,5,10,11
-Dcreate.alfresco.defaults=alfresco,archive -Dnum.shards=12 -Dshard.ids=5,0,11,6
```